api/v1alpha1/trafficmanagerendpoint_types.go

```diff
@@ -0,0 +1,164 @@
+	EndpointRef *TrafficManagerEndpointRef `json:"endpointRef"`
+
+	// +optional
+	// The properties of the endpoint.
+	Properties *TrafficManagerEndpointProperties `json:"properties,omitempty"`
+}
+
+// TrafficManagerEndpointRef is the reference to the endpoint(s).
+// In the multi-cluster scenario, it supports ServiceImport as the endpoint reference, representing a group of endpoints.
+type TrafficManagerEndpointRef struct {
+	// Name is the reference to the ServiceImport in the same namespace of Traffic Manager Profile.
+	// +required
+	Name string `json:"name"`
+}
+
+// TrafficManagerEndpointProperties defines the properties of the Traffic Manager endpoint.
+// For now, only AzureEndpoints are supported.
+// It assumes the service is exposed via PublicIPAddress in Azure and must have a DNS name assigned.
+// Note: To use Traffic Manager with endpoints from other subscriptions, the controller needs to have read access to
+// the endpoint.
+type TrafficManagerEndpointProperties struct {
+	// If Always Serve is enabled, probing for endpoint health will be disabled and endpoints will be included in the traffic
+	// routing method.
+	// +kubebuilder:default=false
+	// +optional
+	AlwaysServe bool `json:"alwaysServe,omitempty"`
+
+	// List of custom headers.
+	// +optional
+	// +kubebuilder:validation:MaxItems=8
+	// Note: using asterisk characters (*) in custom Host headers is unsupported.
+	// Anything applicable for Http and Https protocol overrides custom header settings in profile configuration.
+	CustomHeaders []CustomHeader `json:"customHeaders,omitempty"`
+
+	// The total weight of endpoints behind the serviceImport when using the 'Weighted' traffic routing method.
+	// Possible values are from 1 to 1000.
+	// It is required when the routing method is 'Weighted'.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	// For example, if there are two clusters exporting the service via public ip, each public ip will be configured
+	// as "Weight"/2.
+	Weight *int64 `json:"weight,omitempty"`
+
+	// The fully-qualified DNS name or IP address of the endpoint.
+	// READ-ONLY; populated in the status.
+	// +optional
+	Target *string `json:"target,omitempty"`
+}
+
+// AcceptedEndpoint is the Azure Traffic Manager endpoint which is successfully accepted under the traffic manager Profile.
+type AcceptedEndpoint struct {
+	// Name is a unique Azure Traffic Manager endpoint name generated by the controller.
+	// +required
+	Name *string `json:"name"`
+
+	// +optional
+	Properties *TrafficManagerEndpointProperties `json:"properties,omitempty"`
+
+	// Cluster is where the endpoint is exported from.
+	// +optional
+	Cluster *ClusterStatus `json:"cluster,omitempty"`
+}
+
+type TrafficManagerEndpointStatus struct {
+	// Endpoints contains a list of accepted Azure endpoints which are created or updated under the traffic manager Profile.
+	// +optional
+	Endpoints []AcceptedEndpoint `json:"endpoints,omitempty"`
```

Comment:
the TrafficManagerEndpointSpec has only one ref but the endpoint is an array, why? Is it because one service might have many end points?

Reply:
ServiceImport stands for Services exported in the multi-cluster env. An Azure Traffic Manager endpoint will be created per cluster.